jenkinsapi/job.py

```diff
@@ -64,7 +64,17 @@ def __init__(self, url, name, jenkins_obj):
             'hg': self._get_hg_branch,
             None: lambda element_tree: []
         }
-        self.url = url
+
+        if jenkins_obj.base_server_url() and jenkins_obj.get_use_baseurl():
```

Comment:
This code has been duplicated in `Build()` as well.  Perhaps we could add this logic to the common ancestor `JenkinsBuild`?